### PR TITLE
Remove more unnecessary NuGet package references.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
A continuation of #5242, this PR removes more NuGet packages, some of which were added later.

---

I have seen that some packages like `System.Collections.Immutable` were updated to version 5.0.0. Can I do the same for the other system packages as well?

### CODE REVIEW

**Comment by rainersigwald** [comment]
> I have seen that some packages like `System.Collections.Immutable` were updated to version 5.0.0. Can I do the same for the other system packages as well?

Not trivially, because we often have to coordinate with other repos and Visual Studio itself. I started taking a crack at this in #5910 but got bogged down and probably won't finish before I head out on parental leave.

**Review by Forgind — APPROVED** [review]

**Comment by teo-tsirpanis** [comment]
> I head out on parental leave

Congratulations!! 🥳👶🏻

**Review by BenVillalobos — APPROVED** [review]
LGTM

File: src/Build.UnitTests/Microsoft.Build.Engine.UnitTests.csproj [hunk 1]
```diff
@@ -16,7 +16,6 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="System.Collections.Concurrent" />
```
Comment (BenVillalobos) [inline src/Build.UnitTests/Microsoft.Build.Engine.UnitTests.csproj:19]: Can you also remove the packagereference update in eng\Packages.props?

File: src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj [hunk 2]
```diff
@@ -13,7 +13,6 @@
 
   <ItemGroup>
     <PackageReference Include="System.Security.Principal.Windows" />
-    <PackageReference Include="System.Xml.XPath" />
```
Comment (BenVillalobos) [inline src/Tasks.UnitTests/Microsoft.Build.Tasks.UnitTests.csproj:16]: Can you remove the packagereference update in eng\packages.props for System.Xml.XPath?

File: src/Tasks/Microsoft.Build.Tasks.csproj [hunk 3]
```diff
@@ -1005,8 +1005,6 @@
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'">
     <PackageReference Include="System.CodeDom" />
     <PackageReference Include="System.Reflection.Metadata" />
-    <PackageReference Include="System.Reflection.TypeExtensions" />
```
Comment (BenVillalobos) [inline src/Tasks/Microsoft.Build.Tasks.csproj:1008]: Same as previous comments.

File: src/Tasks/Microsoft.Build.Tasks.csproj [hunk 4]
```diff
@@ -1005,8 +1005,6 @@
...
-    <PackageReference Include="System.Runtime.InteropServices" />
```
Comment (BenVillalobos) [inline src/Tasks/Microsoft.Build.Tasks.csproj:1009]: Same as previous comments.

File: src/Utilities.UnitTests/Microsoft.Build.Utilities.UnitTests.csproj [hunk 5]
```diff
@@ -11,7 +11,6 @@
 
   <ItemGroup>
     <PackageReference Include="Shouldly" />
-    <PackageReference Include="System.IO.FileSystem.Primitives" />
```
Comment (BenVillalobos) [inline src/Utilities.UnitTests/Microsoft.Build.Utilities.UnitTests.csproj:14]: Same as previous.

**Review by BenVillalobos — COMMENTED** [review]
Comments from the PR review meeting: Any `PackageReference` items that were removed should also remove the equivalent `PackageReference` in the `eng\Packages.props` file.

**Comment by teo-tsirpanis** [comment]
Done @BenVillalobos. I also cleaned-up all unused `Packages.props` entries.

**Review by Forgind — APPROVED** [review]
Looks good. Our master branch is temporarily closed to new commits, but we'll merge this soon. Thanks for this!

File: eng/Packages.props [hunk 6]
```diff
@@ -17,32 +17,17 @@
...
     <PackageReference Update="System.Collections.Immutable" Version="5.0.0" />
-    <PackageReference Update="System.Diagnostics.TraceSource" Version="4.0.0" />
-    <PackageReference Update="System.IO.Compression" Version="4.3.0" />
-    <PackageReference Update="System.IO.FileSystem.Primitives" Version="4.3.0" />
-    <PackageReference Update="System.Linq.Parallel" Version="4.0.1" />
     <PackageReference Update="System.Memory" Version="4.5.4" />
-    <PackageReference Update="System.Net.Http" Version="4.3.4" />
     <PackageReference Update="System.Reflection.Metadata" Version="1.6.0" />
-    <PackageReference Update="System.Reflection.TypeExtensions" Version="4.1.0" />
     <PackageReference Update="System.Resources.Extensions" Version="4.6.0" />
```
Comment (Forgind) [inline eng/Packages.props:23]:
```suggestion
    <PackageReference Update="System.Resources.Extensions" Version="4.6.0" />
    <PackageReference Update="System.Runtime.CompilerServices.Unsafe" Version="5.0.0"/>
```

It seems I should not have deleted this one. Will merge if it passes CI this time.

**Comment by Forgind** [comment]
Thanks @teo-tsirpanis!

**Pull request merged**